src/get.brs

```diff
@@ -18,29 +21,69 @@
 '  * value = _.get(data, "a[0]")
 '  * '  => invalid
 '  *
+'  * data = <Component: roSGNode:Group> = {
+'  *   signedIn: false
+'  *   id: "homeScreen"
+'  *   trackingPageInfo: <Component: roAssociativeArray>
+'  *   ...... other common fields of node
+'  * }
+
+'  * value = _.get(data, "signedIn")
+'  *
+'  * value = _.get(data, "0.id") ' value of First child's id
+'  *
+'  * value = _.get(data, "0.1.contentMode") ' value of 1st child of 2nd child's contentMode
+'  *
+'  * value = _.get(data, "trackingPageInfo.pageType")
+'  *
+'
 '  */
 Function rodash_get_(array, path, default=invalid)
 
-if array = invalid or not (type(array) = "roAssociativeArray" or type(array) = "roArray") then return default
+if array = invalid or not (type(array) = "roAssociativeArray" or type(array) = "roArray" or type(array) = "roSGNode") then return default
 
   segments = m.pathAsArray_(path)
   if segments = invalid then return default
 
-  result = invalid
+  result = array
+
   while segments.count() > 0
     key = segments.shift()
-    value = array[key]
-    if value = invalid
-      exit while
-    end if
-    if segments.count() = 0
-      result = value
-      exit while
-    end if
-    if not (type(value) = "roAssociativeArray" or type(value) = "roArray")
-      exit while
+
+    if array <> invalid and type(array) = "roSGNode"
+      
+      if type(key) = "roString" and array.hasField(key)
```

Comment:
When the `array` parameter is node, and the key is a string, we can re-use the existing logic where value = array[key]. The only time we need new logic is if array is a node and key is an integer, then we need to use the getChild() method.